src/focus.service.ts

```diff
@@ -421,10 +380,17 @@ export class FocusService {
 
   public createArcEvent(direction: Direction): ArcEvent {
     const directional = isDirectional(direction);
+    let element: HTMLElement;
+    if (directional) {
+      element = this.findNextFocusByRaycast(direction);
+      if (!element) {
+        element = this.findNextFocus(direction);
```

Comment:
All the elements in the benchmark are next to each other.
+ this has potential to break existing focus behavior. For instance, if the element next to the ref element is slightly below the ref element center, it wont be considered a potential element by "focus ray" even though it is next to the ref element.